bin/configure-fastly.js

```diff
@@ -0,0 +1,176 @@
+
+const PASS_REQUEST_CONDITION_NAME = 'Pass';
+const NOT_PASS_REQUEST_CONDITION_NAME = '!(Pass)'
+const PASS_CACHE_CONDITION_NAME = 'Cache Pass';
+const BUCKET_NAME_HEADER_NAME = 'Bucket name';
+
+var extraAppRoutes = [
+    // Homepage with querystring.
+    // TODO: Should this be added for every route?
+    '^/\\?',
+    // Version output by build
+    '/version\.txt$',
+    // View html
+    '^/[^\/]*\.html'
+];
+
+/*
+ * Given the relative path to the static directory, return an array of
+ * patterns matching the files and directories there.
+ */
+var getStaticPaths = function (pathToStatic) {
+    var staticPaths = glob.sync(path.resolve(__dirname, pathToStatic));
+    return staticPaths.map( function (pathName) {
+        // Reduce absolute path to relative paths like '/js'
+        var base = path.dirname(path.resolve(__dirname, pathToStatic));
+        return '^' + pathName.replace(base, '');
+    });
+}
+
+/*
+ * Given a list of express routes, return a list of patterns to match
+ * the express route and a static view file associated with the route
+ */
+var getViewPaths = function (routes) {
+    return routes.map(function (route) {
+        return route.pattern;
+    });
+}
+
+/*
+ * Given a list of patterns for paths, OR all of them together into one
+ * string suitable for a Fastly condition
+ */
+var pathsToCondition = function (paths) {
+    return paths.reduce(function(conditionString, pattern) {
+        var patternCondition = 'req.url ~ "' + pattern + '"';
+        return conditionString + (conditionString ? ' || ' : '') + patternCondition;
+    }, '');
+}
+
+/*
+ * Combine static paths, routes, and any additional paths to a single
+ * fastly condition to match req.url
+ */
+var getAppRouteCondition = function (pathToStatic, routes, additionalPaths) {
+    var staticPaths = getStaticPaths(pathToStatic);
+    var viewPaths = getViewPaths(routes);
+    var allPaths = [].concat(staticPaths, viewPaths, additionalPaths);
+    return pathsToCondition(allPaths);
+}
+
+var getConditionNameForView = function (view) {
+    return 'routes/' + view;
+};
+
+var getHeaderNameForView = function (view) {
+    return 'rewrites/' + view;
+}
+
+var notPassCondition = {
+    name: NOT_PASS_REQUEST_CONDITION_NAME,
+    statement: getAppRouteCondition('../static/*', routes, extraAppRoutes),
+    type: 'REQUEST',
+    priority: 10
+};
+var passCondition = {
+    name: PASS_REQUEST_CONDITION_NAME,
+    statement: fastly.negateConditionStatement(notPassCondition.statement),
+    type: 'REQUEST',
+    priority: 10
+};
+var passCacheCondition = defaults({name: PASS_CACHE_CONDITION_NAME, type: 'CACHE'}, passCondition);
+var bucketNameHeader = {
+    name: BUCKET_NAME_HEADER_NAME,
+    action: 'set',
+    ignore_if_set: 0,
+    type: 'REQUEST',
+    dst: 'http.host',
+    src: '"' + s3Bucket + '"',
+    request_condition: NOT_PASS_REQUEST_CONDITION_NAME,
+    priority: 1
+};
+
+async.waterfall([
```

Comment:
Not sure about the use of waterfall here. Wouldn't it make more sense for all of this to be in a single `async.auto`?